packages/block-editor/src/components/copy-handler/index.js

```diff
@@ -1,16 +1,88 @@
 /**
  * WordPress dependencies
  */
-import { useRef } from '@wordpress/element';
+import { useCallback, useEffect, useRef } from '@wordpress/element';
 import { serialize, pasteHandler } from '@wordpress/blocks';
-import { documentHasSelection } from '@wordpress/dom';
+import { documentHasSelection, documentHasTextSelection } from '@wordpress/dom';
 import { useDispatch, useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import { getPasteEventData } from '../../utils/get-paste-event-data';
 
+function useFlashBlock() {
+	const { toggleBlockHighlight } = useDispatch( 'core/block-editor' );
+	const timeouts = useRef( [] );
+	const flashBlock = useCallback(
+		( clientId ) => {
+			toggleBlockHighlight( clientId, true );
+			const timeout = setTimeout( () => {
+				toggleBlockHighlight( clientId, false );
+			}, 1000 );
+			timeouts.current.push( timeout );
```

Comment:
This feels a lot like a hack. Does it have its place here? Should the idea of flashing be a proper `block-editor` action? Should there be different primitives for this? cc @youknowriad

Reply:
The `dispatch('core/editor').flashBlock( clientId )` action can also be written as a generator and absorbs the timeout handling to unset the reducer state. When I call `dispatch('core/editor').flashBlock( clientId )` I should expect it to handle all the logic.

Reply:
Thanks, I agree.

One last question: did you namespace it `core/editor` intentionally, as opposed to `core/block-editor`?

Reply:
no, not intentional :) good catch.

Reply:
I pushed f425709c4f, but I don't really like what I did with the introduction of a `SLEEP` control. Do you suggest a different approach, @youknowriad? And if not, should I export `sleep` from `controls`, or keep it private as now?

Reply:
I think the "sleep" control is totally fine. redux-saga, for instance, has a "delay" effect which is equivalent.